frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GraphTableDialog.kt

```diff
@@ -0,0 +1,107 @@
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepositorySnapshot
+import com.virtuslab.gitmachete.frontend.ui.api.table.ISimpleGraphTableProvider
+import java.awt.event.ActionEvent
+import java.util.function.Consumer
+import javax.swing.AbstractAction
+
+class GraphTableDialog
+    private constructor(
+        private val table: JBTable,
+        private val repositorySnapshot: IGitMacheteRepositorySnapshot?,
+        private val dimension: JBDimension,
+        private val okAction: Consumer<IGitMacheteRepositorySnapshot>?,
+        private val editAction: Runnable?,
+        private val cancelButtonVisible: Boolean,
+        private val windowTitle: String,
+        private val okButtonText: String
+    ) : DialogWrapper(/* canBeParent */ false) {
+
+  init {
+    super.init()
+    title = windowTitle
+    setOKButtonText(okButtonText)
+  }
+
+  companion object {
+    // @JvmOverloads annotation instructs the Kotlin compiler to generate overloads
+    // for this function that substitute default parameter values (dimension in our case).
+    @JvmOverloads
+    fun of(
+        gitMacheteRepositorySnapshot: IGitMacheteRepositorySnapshot,
+        windowTitle: String,
+        dimension: JBDimension = JBDimension(/* width */ 800, /* height */ 500),
+        emptyTableText: String?,
+        okAction: Consumer<IGitMacheteRepositorySnapshot>?,
+        editAction: Runnable?,
+        okButtonText: String,
+        cancelButtonVisible: Boolean
+    ) =
+        RuntimeBinding.instantiateSoleImplementingClass(ISimpleGraphTableProvider::class.java)
+            .deriveInstance(gitMacheteRepositorySnapshot, /* toggleListingCommits */ false)
+            .apply { setTextForEmptyTable(emptyTableText ?: "") }
+            .let {
+              GraphTableDialog(
+                  it,
+                  gitMacheteRepositorySnapshot,
+                  dimension,
+                  okAction,
+                  editAction,
+                  cancelButtonVisible,
+                  windowTitle,
+                  okButtonText)
+            }
+
+    fun ofDemoRepository() =
+        RuntimeBinding.instantiateSoleImplementingClass(ISimpleGraphTableProvider::class.java)
+            .deriveDemoInstance()
+            .let {
+          GraphTableDialog(
+              it,
+              repositorySnapshot = null,
+              dimension = JBDimension(/* width */ 800, /* height */ 250),
+              okAction = null,
+              editAction = null,
+              cancelButtonVisible = false,
+              windowTitle = "Git Machete Help",
+              okButtonText = "Close")
+        }
+  }
+
+  override fun createActions() =
+      if (cancelButtonVisible)
+          arrayOf(getEditAction(), getOKAction(), cancelAction).filterNotNull().toTypedArray()
+      else arrayOf(getOKAction())
+
+  private fun getEditAction() =
+      if (editAction == null) null
+      else
+          object : AbstractAction("Edit") {
```

Comment:
Hmm actually that's `Save & Edit` action to be precise (it does the same as `Save` and additionally opens the editor)... not sure if we want to reflect this nuance in the UI

Reply:
right, let it be short but precise